README.md

```diff
@@ -1,2 +1,122 @@
+
+* Prototype of the *Jupyter Workbench*:
+  - Initially, a new page in the existing notebook web-app that allows multiple
+    components on the same browser tab.
+  - Flexible, custimizable tabbed and paneled layout system.
+  - File browser.
+  - Text editor.
+  - Terminal.
+* All new frontend work is being done as small, loosely coupled npm packages.
+* The entire Jupyter Workbench will be a set of plugins (again just npm packages)
+  that exchange runtime APIs.
+* Application infrastucture (keyboard shortcuts, commands, menus, layout), view
+  layer, and other utilities (signals, properties, messages) provided by phosphorjs
+  (http://phosphorjs.github.io/).
+* Models and views that can be used on any web page in the following repos:
+  - https://github.com/jupyter/jupyter-js-services
+  - https://github.com/jupyter/jupyter-js-terminal
+  - https://github.com/jupyter/jupyter-js-editor
+  - https://github.com/jupyter/jupyter-js-filebrowser
+  - https://github.com/jupyter/jupyter-js-input-area
+  - https://github.com/jupyter/jupyter-js-output-area
+  - https://github.com/jupyter/jupyter-js-cells
+  - https://github.com/jupyter/jupyter-js-notebook
+* Wrapping of those models and views as plugins for the Jupyter Workbench
+  - https://github.com/jupyter/jupyter-js-services-plugin
+  - https://github.com/jupyter/jupyter-js-metaservice-plugin
+  - https://github.com/jupyter/jupyter-js-output-area-plugin
+  - https://github.com/jupyter/jupyter-js-notebook-plugin
+  - https://github.com/jupyter/jupyter-js-terminal-plugin
+  - https://github.com/jupyter/jupyter-js-editor-plugin
+  - https://github.com/jupyter/jupyter-js-filebrowser-plugin
+* Movement towards having stable, documented JavaScript APIs.
+* Ability to hide/show code and entire cells based on query syntax.
+
+### Next+2
+
+* New plugins for the Jupyter Workbench:
+  - Standalone output area.
+  - Ability to hook kernels up to text editors and output areas.
+  - Themes.
+
+### Future
+
+* New plugins for the Jupyter Workbench:
+  - Notebook.
+  - Variable inspector.
+  - UI for managing plugins.
+* Dashboarding of output areas and interactive widgets.
+* Replace existing pages in the notebook web-application by the Jupyter Workbench.
+* Interactive debugger.
+* Real-time collaboration on notebooks, text-editors and other plugins.
+* Perform and publish accesibility audit by running an automated tool.
+* Bring our core plugins up to the level of the Web Accessibility Standards
+  (http://www.w3.org/standards/webdesign/accessibility)
+
+## ipywidgets
+
+### Next (4.1)
+
+* Improved and consistent visual style of built-in widgets.
+* Better abstractions for styling widgets using CSS and doing layout with flexbox.
+* Ability to run ipywidgets outside of the notebook.
+* Initial refactoring to start to use npm for packaging the JavaScript code.
+
+### Future
+
+* Further decoupling of the JavaScript and Python kernel backend.
+* Replace views based on backbone.js with phosphorjs widgets.
```

Comment:
Yes, that is why I put this under "Future". It is something that we are thinking seriously about doing "at some point". But also, in my mind, it is likely that we would keep using backbone for model sync, but wrap the backbone models into well-typed interfaces so all of the view classes can be reused outside of ipywidgets by wapping out models that have different backends.  I will add some clarification about this that indicates the uncertainties and questions we have.